assets/js/components/ErrorHandler/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Notification from 'GoogleComponents/notifications/notification';
+
+class ErrorHandler extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			error: null,
+			info: null,
+		};
+	}
+
+	componentDidCatch( error, info ) {
+		global.console.error( 'Caught an error:', error, info );
+
+		this.setState( { error, info } );
+	}
+
+	render() {
+		const { children } = this.props;
+		const { error, info } = this.state;
+
+		// If there is no caught error, render the children components normally.
+		if ( ! error ) {
+			return children;
+		}
+
+		return (
+			<Notification
+				id={ 'googlesitekit-error' }
+				key={ 'googlesitekit-error' }
+				title={ __( 'Site Kit encountered an error', 'google-site-kit' ) }
+				description={ <code>{ error.message }</code> }
+				dismiss={ '' }
```

Comment:
`dismiss` is unnecessary since the notice is not dismissable.

Reply:
Fair enough! This is actually copied from the other various components' `Notification` code, but fair point. Will remove!